app/src/main/java/tech/ula/viewmodel/FilesystemListViewModel.kt

```diff
@@ -78,25 +80,34 @@ class FilesystemListViewModel(private val filesystemDao: FilesystemDao, private
             filesystemUtility.compressFilesystem(filesystem, localTempBackupFile, exportUpdateListener)
 
             if (!localTempBackupFile.exists()) {
-                exportStatusLiveData.postValue(ExportFailure("Exporting to local directory failed"))
+                exportStatusLiveData.postValue(ExportFailure(R.string.error_export_to_local_failed))
                 return@withContext
             }
 
             try {
                 localTempBackupFile.copyTo(externalBackupFile)
                 localTempBackupFile.delete()
             } catch (e: Exception) {
-                exportStatusLiveData.postValue(ExportFailure("Exporting to external directory failed"))
+                exportStatusLiveData.postValue(ExportFailure(R.string.error_export_to_external_failed))
                 localTempBackupFile.delete()
                 return@withContext
             }
 
             when (externalBackupFile.exists() && externalBackupFile.length() > 0) {
                 true -> exportStatusLiveData.postValue(ExportSuccess)
-                false -> exportStatusLiveData.postValue(ExportFailure("Exporting to external directory failed, exported file has no data"))
+                false -> exportStatusLiveData.postValue(ExportFailure(R.string.error_export_to_external_failed_no_data))
             }
         }
     }
+
+    fun startExport(filesystem: Filesystem, activeSessions: List<Session>, filesDir: File) {
+        if (activeSessions.isEmpty()) {
+            val externalDestination = Environment.getExternalStoragePublicDirectory("UserLAnd")
```

Comment:
By using `Environment` at this level, I don't think this function will be unit-testable. I think it's best to pass the `externalDestination` down as a parameter, so that we can use test files and folders as necessary.

Could you also write a test to make sure the behavior of only allowing exports while sessions aren't active doesn't regress?